src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1709,6 +1726,64 @@ export class ConfigureRecommendedExtensionsCommandsContributor extends Disposabl
 			},
 			when: this.workspaceFolderContextKey
 		});
+
+		CommandsRegistry.registerCommand(AddToWorkspaceRecommendationsAction.ID, serviceAccesor => {
+			this.boundAddToRecommendationsContextKey.set(false);
+			const extensionId = (this.editorService.activeEditor as ExtensionsInput).extension.id;
+			serviceAccesor
+				.get(IInstantiationService)
+				.createInstance(
+					AddToWorkspaceRecommendationsAction,
+					AddToWorkspaceRecommendationsAction.ID,
+					AddToWorkspaceRecommendationsAction.LABEL
+				)
+				.run(extensionId)
+				.then(
+					(quickPickCancelled) => this.boundAddToRecommendationsContextKey.set(quickPickCancelled),
+					() => this.boundAddToRecommendationsContextKey.set(this.editorService.activeEditor instanceof ExtensionsInput)
+				);
+		});
+		MenuRegistry.appendMenuItem(MenuId.CommandPalette, {
+			command: {
+				id: AddToWorkspaceRecommendationsAction.ID,
+				title: `${ExtensionsLabel}: ${AddToWorkspaceRecommendationsAction.LABEL}`
+			},
+			when: this.addToRecommendationsContextKey
+		});
+	}
+
+	private updateAddToRecommendationsCommandContextKey(): void {
+		if (!(this.editorService.activeEditor instanceof ExtensionsInput)) {
+			return this.boundAddToRecommendationsContextKey.set(false);
+		}
+
+		const folders = this.workspaceContextService.getWorkspace().folders;
+		const configurationFiles = folders
+			.map(workspaceFolder => workspaceFolder.toResource(paths.join('.vscode', 'extensions.json')));
+		const getFoldersRecommendedExtensionsPromises = configurationFiles.map(configurationFile => {
+			return this.getFolderRecommendedExtensions(configurationFile);
+		});
+		Promise.all(getFoldersRecommendedExtensionsPromises).then(foldersRecommendationsArray => {
+			if (this.editorService.activeEditor instanceof ExtensionsInput) {
+				// Don't show the command if the selected extension is
+				// recommended in ANY the folders of the current workspace
+				const extensionId = this.editorService.activeEditor.extension.id.toLowerCase();
+				const enabled = !foldersRecommendationsArray.some((recommendations: string[]) => {
+					return recommendations.some(r => r.toLowerCase() === extensionId);
```

Comment:
if we change `enabled` to something more specific, the comments above wont be needed. `someWorkspaceReccomends`?

Also, now that this is just a command rather than a button, I could see enabling it whenever any extension folder doesn't have this extension. This does mean you'd need to filter the quick pick or something, so I'd be fine leaving it as is too. Not a big deal.